vision/cifar10/resnet18.jl

```diff
@@ -0,0 +1,163 @@
+
+function make_minibatch(imgs,labels,batch_size)
+  data_set = [(cat(imgs[i]..., dims = 4), labels[:,i])
+          for i in partition(1:length(imgs), batch_size)]
+  return data_set
+end
+
+X = trainimgs(CIFAR10)
+
+train_idxs = 1:49000
+
+train_imgs = [getarray(X[i].img) for i in train_idxs]
+train_labels = Float32.(onehotbatch([X[i].ground_truth.class for i in train_idxs],1:10))
+train_set = make_minibatch(train_imgs,train_labels,batch_size)
+train_data = train_set |>
+  x -> map(y->gpu.(y),x)
+
+verify_idxs = 49001:50000
+verify_imgs = cat([getarray(X[i].img) for i in verify_idxs]..., dims = 4)
+verify_labels = Float32.(onehotbatch([X[i].ground_truth.class for i in verify_idxs],1:10))
+verify_set = [(verify_imgs,verify_labels)]
+verify_data = verify_set |>
+  x -> map(y->gpu.(y),x)
+
+tX = valimgs(CIFAR10)
+test_idxs = 1:10000
+test_imgs = [getarray(tX[i].img) for i in test_idxs]
+test_labels = Float32.(onehotbatch([tX[i].ground_truth.class for i in test_idxs], 1:10))
+test_set = make_minibatch(test_imgs,test_labels,batch_size)
+test_data = test_set |>
+  x -> map(y->gpu.(y),x)
+
+
+
+identity_layer(n) = Chain(Conv((3,3), n=>n, pad = (1,1), stride = (1,1)),
+                                  BatchNorm(n,relu),
+                                  Conv((3,3), n=>n, pad = (1,1), stride = (1,1)),
+                                  BatchNorm(n,relu))
+
+convolution_layer(n) = Chain(Conv((3,3), n=> 2*n, pad = (1,1), stride = (2,2)),
+                             BatchNorm(2*n,relu),
+                             Conv((3,3), 2*n=>2*n, pad = (1,1), stride = (1,1)),
+                             BatchNorm(2*n,relu))
+
+simple_convolution(n) = Chain(Conv((1,1), n=>n, pad = (1,1), stride = (2,2)),
+                              BatchNorm(n,relu))
+
+
+m_filter(n) = Chain(
+  Conv((3,3), n=>2*n, pad = (1,1), stride = (2,2)),
+  BatchNorm(2*n,relu),
+) |> gpu
+
+struct Combinator
+    conv::Chain
+end |> gpu
+Combinator(n) = Combinator(m_filter(n))# |> gpu
+
+
+function (op::Combinator)(x, y)
+  z = op.conv(y)
+  return x + z
+end
+
+
+n = 7
+
+m = Chain(
+  ConvTranspose((n, n), 3 => 3, stride = n),
+  Conv((7,7), 3=>64, pad = (3,3), stride = (2,2)),
+  BatchNorm(64,relu),
+  MaxPool((3,3), pad = (1,1), stride = (2,2)),
+  SkipConnection(identity_layer(64), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(identity_layer(64), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(64), Combinator(64)),
+  SkipConnection(identity_layer(128), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(128), Combinator(128)),
+  SkipConnection(identity_layer(256), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(256), Combinator(256)),
+  SkipConnection(identity_layer(512), (variable_1, variable_2) -> variable_1 + variable_2),
+  MeanPool((7,7)),
+  x -> reshape(x, :, size(x,4)),

```

Comment:
Don't these anonymous functions have names?
```suggestion
  SkipConnection(identity_layer(512), +),
  MeanPool((7,7)),
  Flux.flatten,

```